server/paymentProviders/thegivingblock/webhook.js

```diff
@@ -17,23 +19,33 @@ export async function webhook(req) {
     const pledgeId = payload.pledgeId;
     const valueAtDonationTimeUSD = payload.valueAtDonationTimeUSD;
 
-    let order = await models.Order.findOne({ where: { data: { pledgeId } } });
+    const order = await models.Order.findOne({ where: { data: { pledgeId } } });
     if (!order) {
       throw new Error(`Could not find matching order. pledgeId=${pledgeId}`);
     }
 
     // update totalAmount with latest value
-    order = await order.update({ totalAmount: Number(valueAtDonationTimeUSD) * 100, currency: 'USD' });
+    await order.update({ totalAmount: Number(valueAtDonationTimeUSD) * 100, currency: 'USD' });
+  } else if (req.body.eventType === 'TRANSACTION_CONVERTED') {
+    const pledgeId = payload.pledgeId;
+    const netValueAmount = payload.netValueAmount;
 
-    // process as paid
-    await confirmOrder(order);
+    const order = await models.Order.findOne({ where: { data: { pledgeId } } });
+    if (!order) {
+      throw new Error(`Could not find matching order. pledgeId=${pledgeId}`);
+    }
 
-    order = await order.update({ status: 'PAID' });
+    // update totalAmount with latest value
+    await order.update({ totalAmount: Number(netValueAmount) * 100, currency: 'USD' });
 
-    return;
-  }
+    await order.update({ status: OrderStatus.PAID });
```

Comment:
you can group updates together

Reply:
Ops, totally missed that one. Corrected it. 👍🏽